taf/updater/updater_pipeline.py

```diff
@@ -670,19 +670,65 @@ def validate_out_of_band_and_update_type(self):
     def _validate_last_validated_commit(self, last_validated_commit):
         branch = self.state.users_auth_repo.default_branch
         users_head_sha = self.state.users_auth_repo.top_commit_of_branch(branch)
-        if last_validated_commit != users_head_sha:
```

Comment:
All following checks are only needed if these two commits are not the same, so we should probably keep this. Also, check if the last validated commit is in that git repo's history first, as that should be quicker than running fetch